pkg/api/types.go

```diff
@@ -1202,6 +1208,21 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// LocalVolume represents a local volume accessible only from one node
+type LocalVolumeSource struct {
+	// NodeName represents the node that the volume is located at
+	NodeName string
+	// Fs represents a filesystem-based local volume
+	// +optional
+	Fs *LocalFsVolume
```

Comment:
@saad-ali can you weigh in here? If you are also planning to overload the path for block and file, then I should do the same here to be consistent.